app/src/main/java/de/danoeh/antennapod/fragment/preferences/AutoDownloadPreferencesFragment.java

```diff
@@ -1,29 +1,42 @@
 package de.danoeh.antennapod.fragment.preferences;
 
+import android.Manifest;
 import android.app.Activity;
 import android.content.Context;
+import android.content.pm.PackageManager;
 import android.content.res.Resources;
 import android.net.wifi.WifiConfiguration;
 import android.net.wifi.WifiManager;
+import android.os.Build;
 import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.content.ContextCompat;
 import android.support.v7.preference.CheckBoxPreference;
 import android.support.v7.preference.ListPreference;
 import android.support.v7.preference.Preference;
 import android.support.v7.preference.PreferenceFragmentCompat;
 import android.support.v7.preference.PreferenceScreen;
 import android.util.Log;
-import de.danoeh.antennapod.R;
-import de.danoeh.antennapod.core.preferences.UserPreferences;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+
 public class AutoDownloadPreferencesFragment extends PreferenceFragmentCompat {
     private static final String TAG = "AutoDnldPrefFragment";
+
+    private static final String requestedPermission = Manifest.permission.ACCESS_FINE_LOCATION;
+    private static final int permissionRequestCode = 1;
```

Comment:
Final fields should be named in capital letters. Please also add something like `_LOCATION` to show what it is used for.